pkg/api/controller_test.go

```diff
@@ -200,32 +200,11 @@ func TestRunAlreadyRunningServer(t *testing.T) {
 		conf := config.New()
 		conf.HTTP.Port = port
 
-		ctlr := api.NewController(conf)
-
-		globalDir := t.TempDir()
-
-		ctlr.Config.Storage.RootDirectory = globalDir
-
-		go func() {
-			if err := ctlr.Run(context.Background()); err != nil {
-				return
-			}
-		}()
-
-		// wait till ready
-		for {
-			_, err := resty.R().Get(baseURL)
-			if err == nil {
-				break
-			}
-
-			time.Sleep(100 * time.Millisecond)
-		}
+		ctlr := makeController(conf, t.TempDir(), "")
```

Comment:
test.NewController(...)

In golang world, we New()... everything

Reply:
Thank you for the review, Ram! My first thought for implementing this new controller function was to create it in the test/common.go and use it, but for that, we would need to import conf and api in common.go, and that would cause a cyclic import error since test is imported in api and so on. That's why I implemented it here, to save some lines in controller_test.go. Or are you just suggesting renaming it?